### PR TITLE
[IMP] account: misc improvements of reconciliation models

### PR DESCRIPTION
**Pull request by william-andre**
* Search in Notes and References, not only in the statement's
communication.
* Activate the partner mapping for Suggest Counterpart models
* Always show the Amount Matching field
* Do not block the deletion of models with a partner mapping
* Allow to archive models
* Add a button to get back to Processing state from Validated state on
bank statements.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_bank_statement.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def _get_previous_statement(self):
             # not have any id yet. However if we are updating an existing record, the domain date <= st.date
             # will find the record itself, so we have to add a condition in the search to ignore self.id
             if not isinstance(st.id, models.NewId):
-                domain.append(('id', '!=', st.id))
+                domain.extend(['|', '&', ('id', '<', st.id), ('date', '=', st.date), '&', ('id', '!=', st.id), ('date', '!=', st.date)])
```
Comment (csnauwaert) [inline addons/account/models/account_bank_statement.py:202]: Shouldn't this line that belong to its own commit be merged in saas-13.3 instead of this branch?

Comment (william-andre) [reply addons/account/models/account_bank_statement.py:202]: Yes, I will backport it if you are ok with it but I needed it here (because of the test, but I could change that)

Comment (william-andre) [reply addons/account/models/account_bank_statement.py:202]: Same for the second commit of this branch, both should be backported

**Comment by csnauwaert** [comment]
@robodoo r+ rebase-ff

File: addons/account/models/account_reconcile_model.py [hunk 2]
```diff
@@ -636,64 +623,87 @@ def _get_invoice_matching_query(self, st_lines_with_partner, excluded_ids):
...
+
+        st_ref = " || ' ' || ".join(
+            "COALESCE(%s, '')" % st_ref_name
+            for st_ref_name in st_ref_list
+        )
+        if not st_ref:
+            return "FALSE"
+
+        statement_compare = r"""(
+                {move_field} IS NOT NULL AND substring(REGEXP_REPLACE({move_field}, '[^0-9\s]', '', 'g'), '\S(?:.*\S)*') != ''
```
Comment (nseinlet) [inline addons/account/models/account_reconcile_model.py:652]: just for information, in PostgreSQL, you can use double quotes around table names and field names. I think it's safer.
